solutions/even_or_odd.py

```
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+
+This module provides a function that determines if a number is even or odd
+
+Module contents:
+  - even_or_odd: returns a string indicating if a number is even or odd
+
+Created on 2025/08/06
+@author: ZaidMazen1
+"""
+
+
+# defining the function
+def even_or_odd(num: int) -> str:
+    """Returns a string indicating if a number is even or odd
+
+    Parameters:
+      num (int): The number to be checked
+
+    Returns:
+      str: "Even" if the number is even, "Odd" if the number is odd
+
+    Raises:
+      AssertionError
+        - If num is not an integer
+        - If num is a list
+
```

I dont think you would need the 2nd statement as you only need to raise the first error.

---

I have it so as not to include assertions for more than 1 thing